platforms/core-configuration/configuration-cache/readme.md

```diff
@@ -0,0 +1,95 @@
+# Configuration Cache
+
+## Configuration Cache modes
+
+CC can function in two modes:
+
+- normal mode aka fail-on-problems
+- warning mode aka lenient
+
+Normal mode is how CC functions in the majority of cases, ensuring safety and reliability.
+Warning mode is an adoption and troubleshooting aid, meant to be used only temporarily.
+
+## Problems and build failures
+
+This section gives an overview of how CC problems interact with the build lifecycle,
+when CC checks can result in build failures and how those are reported.
+
+Whenever a problem is reported, it can be
+
+1. **deferred** until the end of the build
+2. **interrupting** the current scope of work
+3. **suppressed** in an incompatible task
+4. **suppressed silently** via graceful degradation
+
+The way each problem is assigned severity depends on the CC mode and the context in which the problem appears.
+See `org.gradle.internal.cc.impl.problems.ProblemSeverity` usage in code.
```

Comment:
💅 All these `See X` notes would be better as links to the classes on GH, or to their javadocs.